pkg/odo/cli/component/delete.go

```diff
@@ -180,6 +181,12 @@ func (do *DeleteOptions) Run() (err error) {
 		}
 	}
 
+	// If request is to delete non existing component exit with exit code 1
+	if !do.isCmpExists && !do.componentDeleteAllFlag {
```

Comment:
Is this a part of the issue we're trying to fix? Besides that, if I try to delete the component that's not pushed I already see this error:
```sh
$ odo delete
 ✗  Component node does not exist on the cluster
```
So, I'm kind of confused as to why we're adding this here.

Reply:
Actually I have removed this case to fix the issue, which was causing one integration test failure for `odo delete`, To fix that i added that. https://github.com/openshift/odo/blob/master/pkg/odo/genericclioptions/context.go#L532

Reply:
The change makes sense. Thanks! :+1: